File: Programming/Practice/27/C++/27/27/27.cpp
#include <iostream>
#include <vector>

using namespace std;

vector<int> Sorting(vector<int> array, int quantity)
{
	// Сортировка пузырьком по убыванию
	int qs = quantity - 1;
	for (int i = 0; i < qs; i++)
	{
		for (int j = 0; j < qs - i; j++)
		{
			if (array[j] < array[j + 1])
			{
				int temp = array[j];
				array[j] = array[j + 1];
				array[j + 1] = temp;
			}
		}
	}
	return array;
}

int main()
{
	setlocale(LC_ALL, "Rus");

	cout << "Введите количество сигалов: " << endl;
	int quantity;
	cin >> quantity;
	vector<int> first(quantity);
	cout << "Введите сигналы: " << endl;
	for (int i = 0; i < quantity; i++)
	{
		cin >> first[i];
	}

	vector<int> last;
	int size = 0;
	for (int i = 0; i < quantity; i++)
	{
		last.push_back(first[i]);
		size++;
		last = Sorting(last, size);

		if (size <= 5)
		{
			//Выводит все элементы (если их не больше 5)
			for (int j = 0; j < size; j++)
			{
				cout << last[j] << " ";
			}
			cout << endl;
		}

		else
		{
			// Цикл выводит только 5 последних и уже отсортированных элементов
			for (int k = size - 5; k < size; k++)
			{
				cout << last[k] << " ";
			}
			cout << endl;
		}
	}

}

File: Programming/Practice/27/Python/27.py
def Sorting(array: list, quantity: int):
	qs = quantity - 1
	for i in range(qs):
		for j in range(qs):
			if (array[j] < array[j + 1]):
				array[j], array[j + 1] = array[j + 1], array[j]
	return array

quantity = int(input("Введите кол-во сигналов: \n"))
first = list(map(int, input('Введите сигналы: \n').split(' ')))
last = []
size = 0

for i in range(quantity):
	last.append(first[i])
	size += 1
	last = Sorting(last, size)

	if size <= 5:
		for j in range(size):
			print(last[j], end=" ")
		print("")

	else:
		k = size - 5
		while k < size:
			print(last[k], end = " ")
			k += 1
		print("")
	i += 1